flink-runtime/src/main/java/org/apache/flink/runtime/blocklist/DefaultBlocklistHandler.java

```diff
@@ -111,26 +111,27 @@ private void assertRunningInMainThread() {
     public void addNewBlockedNodes(Collection<BlockedNode> newNodes) {
         assertRunningInMainThread();
 
-        Collection<BlockedNode> newlyAddedOrMerged = blocklistTracker.addNewBlockedNodes(newNodes);
-        if (!newlyAddedOrMerged.isEmpty()) {
+        BlockedNodeAdditionResult result = blocklistTracker.addNewBlockedNodes(newNodes);
+        Collection<BlockedNode> newlyAddedNodes = result.getNewlyAddedNodes();
+        if (!newlyAddedNodes.isEmpty()) {
             if (log.isDebugEnabled()) {
                 log.debug(
-                        "Newly added/merged {} blocked nodes, details: {}."
+                        "Newly added {} blocked nodes, details: {}."
                                 + " Total {} blocked nodes currently, details: {}.",
-                        newlyAddedOrMerged.size(),
-                        newlyAddedOrMerged,
+                        newlyAddedNodes.size(),
+                        newlyAddedNodes,
                         blocklistTracker.getAllBlockedNodes().size(),
                         blocklistTracker.getAllBlockedNodes());
             } else {
                 log.info(
-                        "Newly added/merged {} blocked nodes. Total {} blocked nodes currently.",
-                        newlyAddedOrMerged.size(),
+                        "Newly added {} blocked nodes. Total {} blocked nodes currently.",
+                        newlyAddedNodes.size(),
                         blocklistTracker.getAllBlockedNodes().size());
             }
 
-            blocklistListeners.forEach(
-                    listener -> listener.notifyNewBlockedNodes(newlyAddedOrMerged));
-            blocklistContext.blockResources(newlyAddedOrMerged);
+            Collection<BlockedNode> allNodes = result.getAllNodes();
```

Comment:
maybe we only need to invoke `blockResources` on `newlyAddedNodes`?

Reply:
Then the `getAllNodes()` is not needed

Reply:
Good idea!